frame/staking/src/lib.rs

```diff
@@ -761,6 +767,102 @@ impl<AccountId, Balance: HasCompact + Zero> UnappliedSlash<AccountId, Balance> {
 	}
 }
 
+/// Something that defines the maximum number of nominations per nominator.
+pub trait NominationsQuota<Balance> {
+	/// Maximum number of nominations. The method `get_quota` may return a larger number of
+	/// nominations than `Self::MaxNominations`. However, `get_quota_safe` returns the bounded
+	/// maximum number of nominations.
+	type MaxNominations: Get<u32>;
+
+	/// Returns the voter's nomination quota within reasonable bounds [`min`, `max`], where `min`
+	/// is 1 and `max` is `Self::MaxNominations`.
+	fn get_quota_safe(balance: Balance) -> u32 {
+		Self::get_quota(balance).max(1).min(Self::MaxNominations::get())
+	}
+
+	// Returns the voter's nomination quota based on the quota implementation.
+	fn get_quota(balance: Balance) -> u32;
```

Comment:
maybe we should collapse `get_quota` and `get_quota_safe` and make sure this call is always safe.

Reply:
Maybe renaming `get_quota` -> `curve` and `get_quota_safe` -> `get_quota` would be the best option here.